src/uu/sort/src/sort.rs

```diff
@@ -364,7 +364,9 @@ fn tokenize_with_separator(line: &str, separator: char) -> Vec<Field> {
             previous_was_separator = false;
         }
     }
-    tokens.last_mut().unwrap().end = line.len();
+    if !previous_was_separator {
+        tokens.last_mut().unwrap().end = line.len();
+    }
```

Comment:
Do you think this would look simpler?

```rust
fn tokenize_with_separator(line: &str, separator: char) -> Vec<Field> {
    let mut tokens = vec![];
    let separator_indices =
        line.char_indices()
            .filter_map(|(i, c)| if c == separator { Some(i) } else { None });
    let mut start = 0;
    for sep_idx in separator_indices {
        tokens.push(start..sep_idx);
        start = sep_idx + 1;
    }
    if start < line.len() {
        tokens.push(start..line.len());
    }
    tokens
}
```

One difference will be in handling empty lines (above will not add any range but currently [0..0] is returned) but that should not cause any issues.

Reply:
This seems in fact simpler, thank you for the suggestion! As you correctly noted, the handling of empty lines is irrelevant (inserting empty ranges at the end of a line is useless in any case)